### PR TITLE
fix: CTA was not actionable for BN only artworks (CBN)

### PR DESCRIPTION
**Pull request by jo-rs**
The type of this PR is: Bugfix

This PR solves [NX-3173]

### Description
The CTA and edition selection on the artwork confirmation modal were only considering `isOfferable` and `isOfferableFromInquiry` for being actionable and displaying availability, respectively. This PR adds a consideration for `isAcquireable` for BN only artworks, and tests to hopefully prevent this issue from resurfacing.

Before:
before

After CTA-only fix:
Screenshot 2022-06-22 at 09 09 53

After full fix:
after

[NX-3173]: https://artsyproduct.atlassian.net/browse/NX-3173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Conversation/Components/Conversation.tsx [hunk 1]
```diff
@@ -54,8 +54,10 @@ const Conversation: React.FC<ConversationProps> = props => {
   const liveArtwork = conversation?.items?.[0]?.liveArtwork
   const artwork = liveArtwork?.__typename === "Artwork" ? liveArtwork : null
 
-  const isOfferable =
-    !!artwork?.isOfferable || !!artwork?.isOfferableFromInquiry
+  const isActionable =
```
Comment (jo-rs) [inline src/v2/Apps/Conversation/Components/Conversation.tsx:57]: Happy to change the name of this variable if something makes more sense

File: src/v2/Apps/Conversation/Components/Conversation.tsx [hunk 2]
```diff
@@ -54,8 +54,10 @@ const Conversation: React.FC<ConversationProps> = props => {
...
+    !!artwork?.isOfferable ||
+    !!artwork?.isOfferableFromInquiry ||
+    !!artwork?.isAcquireable
```
Comment (lean8086) [inline src/v2/Apps/Conversation/Components/Conversation.tsx:60]: Should we use the feature flag for CBN? (not sure though, just asking)

File: src/v2/Apps/Conversation/Components/EditionSelectBox.tsx [hunk 3]
```diff
@@ -25,28 +25,31 @@ export const EditionSelectBox: React.FC<Props> = ({
   selected,
   onSelect,
 }) => {
-  const isOfferable = !!edition.isOfferableFromInquiry || !!edition.isOfferable
+  const isActionable =
+    !!edition.isOfferableFromInquiry ||
+    !!edition.isOfferable ||
+    !!edition.isAcquireable
```
Comment (lean8086) [inline src/v2/Apps/Conversation/Components/EditionSelectBox.tsx:31]: Same question here about if we need the feature flag for `isAcquireable`

**Review by lean8086 — DISMISSED** [review]
LGTM. Though, I left one question about adding the CBN feature flag.

File: src/v2/Apps/Conversation/Components/__tests__/ConfirmArtwork.jest.tsx [hunk 4]
```diff
@@ -187,4 +232,24 @@ describe("Artwork editions", () => {
...
+      expect(radios).toHaveLength(4)
+
+      // Test unavailable edition
+      const lastRadio = radios.pop()
+      expect(lastRadio).toHaveAttribute("disabled")
+      expect(lastRadio).toHaveTextContent("Unavailable")
+
+      // Make sure the rest are not disabled
+      for (const radio of radios) {
+        expect(radio).toBeEnabled()
```
Comment (jo-rs) [inline src/v2/Apps/Conversation/Components/__tests__/ConfirmArtwork.jest.tsx:250]: For whatever reason the linter wants to change `.not.toHaveAttribute("disabled")`, to `.toBeEnabled()`, but they're not equivalent.
So, added the following line `.not.toHaveTextContent("Unavailable")` for safety.

**Pull request merged**